File: Views/DailyChallenge/DailyChallengeCardView.swift
import SwiftUI

struct DailyChallengeCardView: View {
    @EnvironmentObject var appState: AppState
    @StateObject private var savedChallengesService = SavedChallengesService.shared
    
    let challenge: DailyChallenge
    let showDeleteButton: Bool
    let onCompleted: (() -> Void)?
    let onSave: (() -> Void)?
    let onDelete: (() -> Void)?
    
    @State private var isCompleted: Bool = false
    @State private var showSaveConfirmation: Bool = false
    @State private var isAlreadySaved: Bool = false
    
    init(challenge: DailyChallenge, 
         showDeleteButton: Bool = false, 
         onCompleted: (() -> Void)? = nil, 
         onSave: (() -> Void)? = nil, 
         onDelete: (() -> Void)? = nil) {
        self.challenge = challenge
        self.showDeleteButton = showDeleteButton
        self.onCompleted = onCompleted
        self.onSave = onSave
        self.onDelete = onDelete
        self._isCompleted = State(initialValue: challenge.isCompleted)
    }
    
    var body: some View {
        VStack(spacing: 20) {
            // 🎯 Carte magnifique - EXACTEMENT comme FavoriteQuestionCardView
            VStack(spacing: 0) {
                // Header de la carte avec titre (design rose identique aux favoris)
                VStack(spacing: 8) {
                    Text("Love2Love")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.white)
                        .multilineTextAlignment(.center)
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 20)
                .background(
                    LinearGradient(
                        gradient: Gradient(colors: [
                            Color(red: 1.0, green: 0.4, blue: 0.6),
                            Color(red: 1.0, green: 0.6, blue: 0.8)
                        ]),
                        startPoint: .leading,
                        endPoint: .trailing
                    )
                )
                
                // Corps de la carte avec le défi
                VStack(spacing: 30) {
                    Spacer()
                    
                    Text(challenge.localizedText)
                        .font(.system(size: 22, weight: .medium))
                        .foregroundColor(.white)
                        .multilineTextAlignment(.center)
                        .lineSpacing(6)
                        .padding(.horizontal, 30)
                    
                    Spacer(minLength: 20)
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity)
                .background(
                    LinearGradient(
                        gradient: Gradient(colors: [
                            Color(red: 0.2, green: 0.1, blue: 0.15),
                            Color(red: 0.4, green: 0.2, blue: 0.3),
                            Color(red: 0.6, green: 0.3, blue: 0.2)
                        ]),
                        startPoint: .top,
                        endPoint: .bottom
                    )
                )
            }
            .frame(maxWidth: .infinity)
            .frame(height: 400)
            .cornerRadius(20)
            .shadow(color: .black.opacity(0.3), radius: 10, x: 0, y: 5)
            
            // 🎯 Boutons magnifiques - Style identique aux favoris
            if showDeleteButton {
                // Bouton supprimer (style rouge pour danger)
                Button("delete_challenge_button".localized(tableName: "DailyChallenges")) {
                    onDelete?()
                }
                .font(.system(size: 18, weight: .semibold))
                .foregroundColor(.white)
                .frame(maxWidth: .infinity)
                .frame(height: 56)
                .background(Color.red)
                .cornerRadius(28)
            } else {
                // Boutons pour défi du jour (style favoris rose)
                VStack(spacing: 16) {
                    // Bouton "Défi complété"
                    Button(action: {
                        handleChallengeCompleted()
                    }) {
                        HStack {
                            Spacer()
                            
                            Text("challenge_completed_button".localized(tableName: "DailyChallenges"))
                                .font(.system(size: 18, weight: .semibold))
                                .foregroundColor(.white)
                            
                            Image(systemName: isCompleted ? "checkmark.circle.fill" : "circle")
                                .font(.system(size: 18))
                                .foregroundColor(.white)
                                .padding(.leading, 8)
                            
                            Spacer()
                        }
                        .padding(.horizontal, 20)
                        .padding(.vertical, 12)
                    }
                    .frame(maxWidth: .infinity)
                    .background(Color(red: 1.0, green: 0.4, blue: 0.6))
                    .cornerRadius(28)
                    
                    // Bouton "Sauvegarder le défi"
                    Button(action: {
                        handleChallengeSave()
                    }) {
                        HStack {
                            Spacer()
                            
                            Text("save_challenge_button".localized(tableName: "DailyChallenges"))
                                .font(.system(size: 18, weight: .semibold))
                                .foregroundColor(.white)
                            
                            if showSaveConfirmation {
                                Image(systemName: "checkmark.circle.fill")
                                    .font(.system(size: 18))
                                    .foregroundColor(.white)
                                    .padding(.leading, 8)
                                    .scaleEffect(showSaveConfirmation ? 1.2 : 1.0)
                                    .animation(.spring(response: 0.5), value: showSaveConfirmation)
                            } else {
                                Image(systemName: isAlreadySaved ? "bookmark.fill" : "bookmark")
                                    .font(.system(size: 18))
                                    .foregroundColor(.white)
                                    .padding(.leading, 8)
                            }
                            
                            Spacer()
                        }
                        .padding(.horizontal, 20)
                        .padding(.vertical, 12)
                    }
                    .frame(maxWidth: .infinity)
                    .background(Color(red: 1.0, green: 0.4, blue: 0.6))
                    .cornerRadius(28)
                }
            }
        }
        .onAppear {
            // Vérifier si le défi est déjà sauvegardé
            checkIfAlreadySaved()
        }
        .onChange(of: savedChallengesService.savedChallenges) { _ in
            // Re-vérifier quand la liste change
            checkIfAlreadySaved()
        }
    }
    
    // MARK: - Actions
    
    private func handleChallengeCompleted() {
        let newCompletionState = !isCompleted
        
        withAnimation(.spring(response: 0.6)) {
            isCompleted = newCompletionState
        }
        
        // Persister l'état dans Firebase et cache
        if newCompletionState {
            DailyChallengeService.shared.markChallengeAsCompleted(challenge)
        } else {
            DailyChallengeService.shared.markChallengeAsNotCompleted(challenge)
        }
        
        onCompleted?()
    }
    
    private func handleChallengeSave() {
        if !isAlreadySaved {
            onSave?()
            showSaveConfirmation = true
            isAlreadySaved = true
            
            // Reset après 3 secondes
            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
                showSaveConfirmation = false
            }
        }
    }
    
    private func checkIfAlreadySaved() {
        isAlreadySaved = savedChallengesService.isChallengeAlreadySaved(challenge)
    }
}

// MARK: - Preview

#Preview {
    VStack(spacing: 20) {
        DailyChallengeCardView(
            challenge: DailyChallenge(
                challengeKey: "daily_challenge_1",
                challengeDay: 1,
                scheduledDate: Date(),
                coupleId: "test_couple"
            ),
            showDeleteButton: false,
            onCompleted: {
                print("Challenge completed")
            },
            onSave: {
                print("Challenge saved")
            }
        )
        .environmentObject(AppState())
        
        DailyChallengeCardView(
            challenge: DailyChallenge(
                challengeKey: "daily_challenge_2",
                challengeDay: 2,
                scheduledDate: Date(),
                coupleId: "test_couple",
                isCompleted: true
            ),
            showDeleteButton: true,
            onDelete: {
                print("Challenge deleted")
            }
        )
        .environmentObject(AppState())
    }
    .padding()
    .background(Color(.systemGroupedBackground))
}

File: Views/DailyChallenge/DailyChallengeCardView.kt
package com.love2loveapp.core.ui.views.dailychallenge

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Bookmark
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.love2loveapp.R
import com.love2loveapp.core.ui.extensions.ChatText
import com.love2loveapp.core.viewmodels.AppState
import com.love2loveapp.model.DailyChallenge

/**
 * Carte d'affichage d'un défi du jour
 * Équivalent Kotlin Compose du DailyChallengeCardView Swift
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun DailyChallengeCardView(
    challenge: DailyChallenge,
    showDeleteButton: Boolean,
    onCompleted: () -> Unit,
    onSave: () -> Unit,
    appState: AppState,
    modifier: Modifier = Modifier,
    onDelete: (() -> Unit)? = null
) {
    var isCompleted by remember { mutableStateOf(false) }
    
    Card(
        modifier = modifier
            .fillMaxWidth()
            .shadow(
                elevation = 8.dp,
                shape = RoundedCornerShape(24.dp)
            ),
        shape = RoundedCornerShape(24.dp),
        colors = CardDefaults.cardColors(containerColor = Color.White)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(24.dp),
            verticalArrangement = Arrangement.spacedBy(20.dp)
        ) {
            // En-tête avec jour du défi
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "${stringResource(R.string.challenge_day)} ${challenge.challengeDay}",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color(0xFFFD267A)
                )
                
                if (showDeleteButton && onDelete != null) {
                    IconButton(
                        onClick = onDelete,
                        modifier = Modifier.size(24.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Delete,
                            contentDescription = "Supprimer",
                            tint = Color.Red,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
            }
            
            // Contenu du défi
            Column(
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                Text(
                    text = challenge.challengeText,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color.Black,
                    lineHeight = 24.sp
                )
                
                if (challenge.challengeDescription?.isNotBlank() == true) {
                    Text(
                        text = challenge.challengeDescription,
                        fontSize = 14.sp,
                        color = Color.Black.copy(alpha = 0.7f),
                        lineHeight = 20.sp
                    )
                }
            }
            
            // Boutons d'action
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                // Bouton "Terminé"
                Button(
                    onClick = {
                        isCompleted = true
                        onCompleted()
                    },
                    modifier = Modifier.weight(1f),
                    enabled = !isCompleted,
                    colors = ButtonDefaults.buttonColors(
                        containerColor = if (isCompleted) Color.Green else Color(0xFFFD267A),
                        disabledContainerColor = Color.Green
                    ),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.Check,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                        
                        Text(
                            text = if (isCompleted) {
                                stringResource(R.string.challenge_completed)
                            } else {
                                stringResource(R.string.mark_as_completed)
                            },
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Medium
                        )
                    }
                }
                
                // Bouton "Sauvegarder"
                OutlinedButton(
                    onClick = onSave,
                    modifier = Modifier.weight(1f),
                    colors = ButtonDefaults.outlinedButtonColors(
                        contentColor = Color(0xFFFD267A)
                    ),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Row(
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Icon(
                            imageVector = Icons.Default.Bookmark,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp)
                        )
                        
                        Text(
                            text = stringResource(R.string.save_challenge),
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Medium
                        )
                    }
                }
            }
        }
    }
}